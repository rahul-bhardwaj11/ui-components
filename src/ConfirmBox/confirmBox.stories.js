import React from 'react';
import { storiesOf } from '@storybook/react';
import ConfirmBox from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

const stories = storiesOf('ConfirmBox', module);
stories.addDecorator(withKnobs);

stories.add(
  'ConfirmBox',
  withInfo('Basic usage ConfirmBox')(() => (
    <ConfirmBox title="Are your sure you want to delete ?" placement="bottom">
      <a href="#">Delete</a>
    </ConfirmBox>
  ))
);
