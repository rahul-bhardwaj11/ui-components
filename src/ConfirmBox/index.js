import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import AntPopconfirm from 'antd/lib/popconfirm';
import 'antd/lib/popconfirm/style/css';
import styled from 'styled-components';
import theme from '../styles/theme';
import mixins from '../styles/mixins';

const StyledPopconfirm = styled.div`
  font-family: inherit;
  .ant-popover {
    z-index: ${mixins.zIndex.CONFIRM_BOX};
  }
  .ant-popover-message-title {
    ${mixins.blackLink()};
    padding-left: 0px;
  }
  .ant-popover-inner-content {
    padding: 24px;
  }
  .ant-popover-message > .anticon {
    display: none;
  }
  .ant-popover-buttons {
    margin-bottom: 0px;
    .ant-btn {
      ${mixins.textBtn()};
      ${mixins.button()};
      color: ${theme.colors.OUTER_SPACE};
      min-width: 75px;

      &.ant-btn-primary {
        ${mixins.primaryBtn()};
        ${mixins.button()};
        color: #fff;
        &:hover {
          ${mixins.primaryBtnHover()};
          color: #fff;
        }
        &:focus,
        &.active {
          ${mixins.primaryBtnHover()};
          color: #fff;
        }
      }
      &:hover,
      &:focus {
        ${mixins.textBtn()};
        color: ${theme.colors.ICON};
      }
    }
  }
`;

class ConfirmBox extends Component {
  static propTypes = {
    children: PropTypes.node,
    placement: PropTypes.string,
    getPopupContainer: PropTypes.func
  };
  confirmContainer = null;
  element = null;
  constructor(p) {
    super(p);
    this.element = document.createElement('div');
  }

  componentDidMount() {
    document.body.appendChild(this.element);
  }

  componentWillUnmount() {
    document.body.removeChild(this.element);
  }

  render() {
    let { children, getPopupContainer } = this.props;
    const container =
      this.props.getPopupContainer && this.props.getPopupContainer();
    if (!getPopupContainer || getPopupContainer()) {
      return (
        <React.Fragment>
          {ReactDOM.createPortal(
            <StyledPopconfirm
              innerRef={e => e && (this.confirmContainer = e)}
            />,
            container || this.element
          )}
          <AntPopconfirm
            {...this.props}
            getPopupContainer={() => {
              return this.confirmContainer;
            }}
          >
            {children}
          </AntPopconfirm>
        </React.Fragment>
      );
    } else {
      return null;
    }
  }
}

export default ConfirmBox;
