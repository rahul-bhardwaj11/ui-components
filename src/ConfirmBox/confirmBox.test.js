import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmBox from './index';

describe('ConfirmBox', () => {
  it('matches default snapshot', () => {
    const tree = renderer
      .create(<ConfirmBox title="This is test title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
