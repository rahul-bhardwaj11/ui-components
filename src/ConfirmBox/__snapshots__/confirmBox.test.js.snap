// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmBox matches default snapshot 1`] = `
<span
  onBlur={undefined}
  onClick={[Function]}
  onContextMenu={undefined}
  onFocus={undefined}
  onMouseDown={[Function]}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onTouchStart={[Function]}
/>
`;
